frontend/app/.server/routes/helpers/renew-child-route-helpers.ts

```diff
@@ -171,16 +174,20 @@ export function validateRenewChildStateForReview({ params, state }: ValidateStat
     throw redirect(getPathById('public/renew/$id/child/confirm-address', params));
   }
 
-  if (hasAddressChanged && addressInformation === undefined) {
-    throw redirect(getPathById('public/renew/$id/child/update-address', params));
+  if (hasAddressChanged && isHomeAddressSameAsMailingAddress && mailingAddress === undefined) {
+    throw redirect(getPathById('public/renew/$id/child/update-mailing-address', params));
+  }
+
+  if (hasAddressChanged && !isHomeAddressSameAsMailingAddress && homeAddress === undefined) {
+    throw redirect(getPathById('public/renew/$id/child/update-mailing-address', params));
```

Comment:
I might be wrong here, but what if we make the two `if` statements into this:

`if (hasAddressChanged && isHomeAddressSameAsMailingAddress && mailingAddress === undefined){}`

Reply:
I agree with your comment that we can just do `if (hasAddressChanged && mailingAddress === undefined)`, since if the applicant's address changed, they must enter a mailing address. 

I think that the second condition should redirect to `/update-home-address` instead.

Reply:
Removed the "isHomeAddressSameAsMailingAddress" from the first conditional and fixed the redirect on the second one to target the home address page.